### ISSUE TITLE
Documentation CI job not failing as expected

### ISSUE THREAD

**Issue by hvy** [optuna/optuna#818]
The documentation job in our CI passes even then errors are raised as they are relaxed to warnings. See https://github.com/optuna/optuna/issues/817. We should make sure such failures are reported as errors.

**Comment by harupy** [comment]
@hvy 

Adding `-W` to `SPHINXOPTS` turns warnings into errors.

```diff
# You can set these variables from the command line.
- SPHINXOPTS    =
+ SPHINXOPTS    = -W
SPHINXBUILD   = sphinx-build
SPHINXPROJ    = Optuna
SOURCEDIR     = source
BUILDDIR      = build
```

https://www.sphinx-doc.org/en/master/man/sphinx-build.html

**Comment by harupy** [comment]
Can I make a PR to fix all the existing warnings and add `-W` to `SPHINXOPTS`?

**Comment by hvy** [comment]
Highly appreciated. I also verified you change with `SPHINXOPTS` to address this issue. Thanks!

**Comment by hvy** [comment]
Closing as it's fixed by #819.
